gsy_framework/schema/validators.py

```diff
@@ -49,6 +56,128 @@ def validate(self, data, raise_exception: bool = False) -> (bool, str):
             return False, str(exc)
 
 
+class AVROSchemaSerializer(AVROSchemaValidator):
+    """Serializer class that uses AVRO schemas for serializing and deserializing dicts to bytes."""
+    # pylint: disable=no-self-use
+
+    def modify_input(self, data: Dict) -> Dict:
+        """Modify the input data dict in order for AVRO to properly serialize it."""
+        return data
+
+    def modify_output(self, data: Dict) -> Dict:
+        """Modify the output data dict in order for AVRO  to properly deserialize it."""
+        return data
+
+    def serialize(self, data: Dict, raise_exception: bool) -> bytes:
+        """Validate and serialize dictionary."""
+        data = self.modify_input(data)
+        is_valid, errors = self.validate(data, raise_exception)
+        if not is_valid:
+            raise GSySerializationException(f"Failed to serialize data: {errors}")
+        writer = avro.io.DatumWriter(self.schema)
+        bytes_writer = BytesIO()
+        encoder = avro.io.BinaryEncoder(bytes_writer)
+        writer.write(data, encoder)
+        return bytes_writer.getvalue()
+
+    def deserialize(self, data: bytes) -> Dict:
```

Comment:
I do not see these serializing and deserializing methods being used on the gsy-web or gsy-e side. I guess this is for the future, when you plan to exchange pickling end unpickling the payload with these methods, right?

Reply:
Exactly, essentially the class `DataSerializer` needs to be removed and replaced with the serialize / deserialize methods in the long run. I thought that since we already use AVRO for validation, why not use it for serialization too and remove one extra class that we do not need meanwhile.